module/gui/Tools.py

```diff
@@ -0,0 +1,518 @@
+            if noTranslation:
+                title += " - " + unicode("Error")
+            else:
+                title += " - " + _("Error")
+        withIcon = not tmpIcon.isNull()
+        if withIcon:
+            self.iconLabel.setPixmap(tmpIcon.pixmap(iconSize, iconSize))
+            self.iconLabel.adjustSize()
+            self.iconLabel.setFixedSize(self.iconLabel.size())
+        else:
+            self.iconLabel.setFixedSize(0, 0)
+        self.setWindowTitle(title)
+
+        # constants
+        minWidth = 320  # minimum dialog width
+        maxWidth = 1000 # maximum dialog width
+        addWidth = 20   # extra width
+        icon_x = 0
+        icon_y = 0
+        text_x = 13
+        if not withIcon:
+            text_x += 7
+        text_y = 0
+        buttonBox_v_spacer = 10
+
+        # QTextEdit
+        self.textEdit = QTextEdit()
+        self.textEdit.setReadOnly(True)                                             # read-only
+        self.textEditPal = QPalette(self.textEdit.palette())
+        newcolor = QColor(QLabel().palette().color(QPalette.Active, QLabel().backgroundRole()))
+        self.textEditPal.setColor(QPalette.Base, newcolor)
+        self.textEdit.setPalette(self.textEditPal)                                  # set background color
+        self.textEdit.setFocusPolicy(Qt.NoFocus)                                    # remove from taborder
+        self.textEdit.setTextInteractionFlags(Qt.NoTextInteraction)                 # prohibit text selection
+        self.textEdit.setContextMenuPolicy(Qt.NoContextMenu)                        # disable context menu
+        self.textEdit.setFrameStyle(QFrame.NoFrame)                                 # remove frame
+        self.textEdit.setVerticalScrollBarPolicy(Qt.ScrollBarAlwaysOff)             # no vertical scrollbar
+        self.textEdit.setHorizontalScrollBarPolicy(Qt.ScrollBarAlwaysOff)           # no horizontal scrollbar
+        self.textEdit.setLineWrapMode(QTextEdit.NoWrap)
+        self.textEdit.setText(text)
+        self.textEdit.setFixedWidth(self.textEdit.document().size().width() + self.textEdit.contentsMargins().left() + self.textEdit.contentsMargins().right())
+        self.textEdit.setFixedHeight(self.textEdit.document().size().height() + self.textEdit.contentsMargins().top() + self.textEdit.contentsMargins().bottom())
+
+        # buttons
+        self.buttonBox = QDialogButtonBox()
+        if btnSet == "OK":
+            self.okBtn = self.buttonBox.addButton(QDialogButtonBox.Ok)
+            if noTranslation:
+                self.buttonBox.button(QDialogButtonBox.Ok).setText(unicode("OK"))
+            else:
+                self.buttonBox.button(QDialogButtonBox.Ok).setText(_("OK"))
+            self.connect(self.okBtn, SIGNAL("clicked()"), self.accept)
+        elif btnSet == "OK_CANCEL":
+            self.okBtn     = self.buttonBox.addButton(QDialogButtonBox.Ok)
+            self.cancelBtn = self.buttonBox.addButton(QDialogButtonBox.Cancel)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QDialogButtonBox' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)